php/libraries/Password.class.inc

```diff
@@ -200,6 +200,11 @@ class Password
      */
     public final function __construct(string $value)
     {
+        // If a password contains a special character such as '&' or '<' then
+        // its possible that it has been encoded during transmission to the
+        // class.
+        $value = htmlspecialchars_decode($value);
```

Comment:
This is mixing concerns.. the Password class shouldn't know anything about HTML. The right place to do this is the caller before the constructor is called.